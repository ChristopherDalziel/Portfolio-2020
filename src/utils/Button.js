import React from "react";
import styled from "@emotion/styled";

const StyledButton = styled.button`
  font-family: "Open Sans", sans-serif;
  background-color: #ef636c;
  width: 100%;
  border: none;
  color: white;
  padding: 12px 15px;
  cursor: pointer;
  font-size: 15px;
  &:hover {
    background: #a8b7ab;
    -webkit-transition: background 0.3s ease-in-out;
    -moz-transition: background 0.3s ease-in-out;
    transition: background-color 0.3s ease-in-out;
  }
`;

const Button = ({ type = "button", handleClick, children }) => {
  return (
    <StyledButton type="true" onClick={handleClick}>
      {children}
    </StyledButton>
  );
};

export default Button;
