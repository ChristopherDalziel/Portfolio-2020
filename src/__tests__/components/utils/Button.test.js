import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Button from "../../../utils/Button";

test("Button should render with passed props", () => {
  const onClick = () => {
    // dummy for test
  };
  const wrapper = shallow(
    <Button type="submit" handleClick={onClick}>
      Test button
    </Button>
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});
