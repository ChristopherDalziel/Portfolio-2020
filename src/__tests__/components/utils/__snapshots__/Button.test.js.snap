// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render with passed props 1`] = `
<Styled(button)
  onClick={[Function]}
  type={true}
>
  Test button
</Styled(button)>
`;
